ratatui/src/widgets/list/list.rs

```diff
@@ -290,13 +293,13 @@ impl<'a> List<'a> {
     /// # Examples
     ///
     /// ```rust
-    /// use ratatui::widgets::List;
+    /// use ratatui::{text::Line, widgets::List};
     ///
     /// let items = ["Item 1", "Item 2"];
-    /// let list = List::new(items).highlight_symbol(">>");
+    /// let list = List::new(items).highlight_symbol(Line::from(">>"));
     /// ```
     #[must_use = "method moves the value of self and returns the modified value"]
-    pub const fn highlight_symbol(mut self, highlight_symbol: &'a str) -> Self {
+    pub fn highlight_symbol(mut self, highlight_symbol: Line<'a>) -> Self {
         self.highlight_symbol = Some(highlight_symbol);
         self
     }
```

Comment:
This function should take `Into<Line>` for allowing flexible conversion between types :)

```suggestion
    pub fn highlight_symbol<T>(mut self, highlight_symbol: T) -> Self
    where
        T: Into<Line<'a>>,
    {
        self.highlight_symbol = Some(highlight_symbol.into());
        self
    }
```

Reply:
Added in next commit: https://github.com/ratatui/ratatui/pull/1467/commits/e3ed0feac72fafe1534ad6fdd21444ff6af2f013